### DISCUSSION TITLE
Question about incrementing Prestashop order IDs [PS1.6]

### DISCUSSION THREAD

**Discussion by BrunoLcrt** [PrestaShop/PrestaShop#31003] [Q&A] [answered]
Question sur l'incrémentation des ID de commande Prestashop :
Bonjour à tous et merci pour vos réponses éventuelles !
Quelqu'un pourrait-il me dire où le script prend-il l'ID de référence pour créer le nouvel id_order incrémenté du nouvel enregistrement (la nouvelle commande) ?
Visiblement, ce n'est pas l'id_order du dernier enregistrement...
---
Question about incrementing Prestashop order IDs:
Hi everyone and thanks for any answers!
Could someone tell me where the script takes the reference ID to create the new incremented order_id of the new record (the new order)?
Obviously, it's not the id_order of the last record...
---
Bruno

**Comment by kpodemski** [comment]
Hi Bruno,

It's in the `Order::generateReference()` :)

**Reply by BrunoLcrt** [reply]
Bonjour Krystian et merci de vous pencher sur mon problème !
En fait je ne comprends pas votre réponse.
Pourriez-vous développer un peu plus ?
(j'interviens sur une version multi-boutiques 1.6.0.9)
---
Hello Krystian and thank you for looking into my problem!
Actually I don't understand your answer.
Could you elaborate a bit more?
(I am working on a multi-shop version 1.6.0.9)

**Reply by kpodemski** [reply]
The function that generates the reference is inside classes/order/Order.php in `generateReference` :)

**Comment by BrunoLcrt** [comment]
Bonjour Krystian et merci de prendre de votre temps pour m'aider à solutionner ma recherche !
En effet dans order.php on trouve comment est construit le champ « référence » de la table ps_orders mais c'est le champ « id_order » que je recherche...
Mais sa fonction « getNextOrderId() » est juste au-dessus !
Ça confirme bien que l'id_order est créé à cette étape, mais ça ne solutionne pas mon problème.
Est-il possible que le script renvoie une réponse perturbée par le cache ?
Si je modifie l'id_order du dernier enregistrement (et tous les liens en rapport avec cette commande dans les autres tables bien sûr), le prochain enregistrement créé ne tient pas compte de cet id_order modifié et il reprend l'ancien id_order que j'ai remplacé !
Comment faire ? :)
Merci de votre patience

---

Hello Krystian and thank you for taking your time to help me solve my problem!
Indeed in order.php we find how the field "reference" of the table ps_orders is built but it is the field "id_order" that I am looking for...
But its function "getNextOrderId()" is just above!
This confirms that the id_order is created at this stage, but it doesn't solve my problem.
Is it possible that the script is returning a response that is confused by the cache?
If I change the order_id of the last record (and all the links related to this command in the other tables of course), the next record created does not take into account this modified order_id and it takes the old order_id that I replaced!
How do I do this? :)
Thanks for your patience

**Reply by kpodemski** [reply]
Yes, it may be related to the cache. If you use some third-party modules, then they use records from `information.schema` sometimes, and that won't work on MySQL 8 by default, but I guess you're not using 1.6 with MySQL 8 :)
Do you use any additional cache on your server? Redis? Memcached?

Could you provide me the screenshot of how it looks from your back office? You have:

Order 11, Reference 10
Order 10, Reference 9
?

**Comment by BrunoLcrt** [comment]
No the cache is not activated
https://i.gyazo.com/96531a89816be4a8d476899b2bdde214.png

**Comment by BrunoLcrt** [comment, marked as answer]
Si je remplace l'id_order 91439 par 9179 (et dans toutes les autres tables liées), le prochain id_order généré par le script sera 91440 !

-

If I replace the order_id 91439 with 9179 (and in all other related tables), the next order_id generated by the script will be 91440!

**Reply by kpodemski** [reply]
Oh, ok :) It's because you have to change the value of `auto_increment` field in your database table. Tutorial:
https://www.tutorialspoint.com/change-the-auto-increment-counter-in-mysql

Please be careful, it's risky if not done right. If you get some IDs duplicated, you'll have a lot of issues!

**Comment by BrunoLcrt** [comment]
Ok, on ne voit pas la valeur stockée alors ?
Je dois lancer la commande sql depuis la table ps_orders :
alter table ps_orders auto_increment=9179;
pour le dernier enregistrement ou bien 9180 pour le prochain ?

---

 Ok, we don't see the stored value then?
I need to run the sql command from the ps_orders table:
alter table ps_orders auto_increment=9179;
for the last record or 9180 for the next one?

**Comment by BrunoLcrt** [comment]
J'ai choisi 9180 et...
Ça fonctionne !
alter table `ps_orders` auto_increment=9180;
You know what? I'm happy!

Je suis très fier de vous Krystian, thanks a lot!

---

I chose 9180 and...
It works!
alter table `ps_orders` auto_increment=9180;
You know what? I'm happy!

I am very proud of you Krystian, thanks a lot!


https://www.youtube.com/watch?v=3AA7Ub1c2cI
